### PR TITLE
Improved verbiage/documentation throughout the project

### PR DESCRIPTION
**Pull request by charlesdaniels**
* README has been improved considerably
* Help/usage messages have been improved 
* BitShuffle now follows a defined versioning convention (semver)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -20,41 +20,147 @@ AppVeyor:
...
+(I recommend symlinking ``scripts/pre-commit`` to ``.git/hooks/``).
+
+If you would like to contribute by sending patches over e-mail, that is fine
+too, just get in touch with `@charlesdaniels <https://github.com/charlesdaniels>`_.
+
+Technical Details
+=================
+
+
+BitShuffel Data Packet Specification (compatibilty level 1)
```
Comment (jyn514) [inline README.rst]: typo in bitshuffle

File: bitshuffle/bitshuffle.py [hunk 2]
```diff
@@ -140,30 +142,39 @@ def encode_file(fhandle, chunksize, compresslevel, compresstype):
 
 
 def main():
-    parser = argparse.ArgumentParser(description="")
+    parser = argparse.ArgumentParser(description="A tool for encoding and " +
+                                     "decoding arbitrary binary data to " +
+                                     "ASCII test")
```
Comment (jyn514) [inline bitshuffle/bitshuffle.py]: typo in 'text'

**Review by jyn514 — CHANGES_REQUESTED** [review]
Looks good. Couple typos. Also note that I've added commits after you submitted the request.

File: README.rst (hunk 1 above)
Comment (charlesdaniels) [reply README.rst]: fixed in 0a23b02

File: bitshuffle/bitshuffle.py (hunk 2 above)
Comment (charlesdaniels) [reply bitshuffle/bitshuffle.py]: fixed in 0a23b02

**Comment by charlesdaniels** [comment]
Everything should be fixed as of 0a23b02 - let me know

**Review by jyn514 — APPROVED** [review]

**Pull request merged**